java/fury-core/src/main/java/org/apache/fury/serializer/EnumSerializer.java

```diff
@@ -40,20 +50,54 @@ public EnumSerializer(Fury fury, Class<Enum> cls) {
       Preconditions.checkArgument(enclosingClass.isEnum());
       enumConstants = enclosingClass.getEnumConstants();
     }
+
+    if (fury.getConfig().serializeEnumByName()) {
+      // as we know the size of enum is fixed, initialize the size of map with that value
+      int initialCapacity = (int) Math.ceil(enumConstants.length / 0.99f);
+
+      metaStringtoEnumRepresentation = new HashMap<>(initialCapacity, 0.99f);
+
+      for (Enum enumConstant : enumConstants) {
+        if (enumConstant != null) {
+          MetaString ms =
+              Encoders.GENERIC_ENCODER.encode(enumConstant.name(), MetaString.Encoding.UTF_8);
+          MetaStringBytes msb = metaStringResolver.getOrCreateMetaStringBytes(ms);
+          metaStringtoEnumRepresentation.put(msb, enumConstant);
+        }
+      }
+    } else {
+      metaStringtoEnumRepresentation = null;
+    }
   }
 
   @Override
   public void write(MemoryBuffer buffer, Enum value) {
-    buffer.writeVarUint32Small7(value.ordinal());
+    if (fury.getConfig().serializeEnumByName()) {
+      MetaString enumMetaString =
```

Comment:
If we create an array MetaStringBytes[] whose index are enum oridinal, we could get MetaStringBytes just from the oridinal. maybe this will make IdentifyMap work

Reply:
do you mean like this?
it's still not able to find the key for identity map.
im no expert, but seems like because it doesnt refer to the same object while checking the key.
while it works if we use HashMap.

```
      metaStringBytesArr = new MetaStringBytes[enumConstants.length];
      for (Enum enumConstant : enumConstants) {
        if (enumConstant != null) {
          MetaString ms =
              Encoders.GENERIC_ENCODER.encode(enumConstant.name(), MetaString.Encoding.UTF_8);
          MetaStringBytes msb = metaStringResolver.getOrCreateMetaStringBytes(ms);
          metaStringtoEnumRepresentation.put(msb, enumConstant);
          metaStringBytesArr[enumConstant.ordinal()] = msb;
        }
      }
```
```
  @Override
  public void write(MemoryBuffer buffer, Enum value) {
    if (fury.getConfig().serializeEnumByName()) {
      MetaStringBytes metaStringBytes = metaStringBytesArr[value.ordinal()];
      metaStringResolver.writeMetaStringBytes(buffer, metaStringBytes);
    } else {
      buffer.writeVarUint32Small7(value.ordinal());
    }
  }
```
```
  @Override
  public Enum read(MemoryBuffer buffer) {
    if (fury.getConfig().serializeEnumByName()) {
      MetaStringBytes metaStringBytes = metaStringResolver.readMetaStringBytes(buffer);
      Enum e = metaStringtoEnumRepresentation.get(metaStringBytes);
      if (e != null) {
        return e;
      }
      return handleNonexistentEnumValue(metaStringBytes.decode(Encoders.GENERIC_DECODER));
    } else {
      int value = buffer.readVarUint32Small7();
      if (value >= enumConstants.length) {
        return handleNonexistentEnumValue(value);
      }
      return enumConstants[value];
    }
  }
```

Reply:
metaStringResolver may return a different object, we could use fury ObjectMap instead.  fury ObjectMap is a linear probe based map, which is faster than hashmap for smaller map size